### PR TITLE
Ref #7842 - Test can not be published with empty item(s)

### PR DESCRIPTION
**Pull request by k-v-vladimir**
Ref #7842 - Test can not be published with empty item(s)
https://oat-sa.atlassian.net/browse/TAO-7842
Steps to reproduce: 
1. Log in as admin; 
2. Create some items, one of them should be empty; 
3. Create a test with created items including empty one; 
4. Publish created test;

Expected result: Error message should display when trying to publish test. 
Actual result: Test published and can be runned by TT

The test items are checked for emptiness. If there are no elements, the error is generated and sent to the client.

### CODE REVIEW

**Comment by ferenckiss89** [comment]
- Branch has to be always created from develop branch and the PR should go back to develop (not master)
- Please create a description for the PR (jira ticket link + list of the changes, and a guide for the reviewers if needed)
- When you update an extension the manifest.php has to be updated always with the new version. In this case from 8.2.1 -> 8.2.2 (semantic versioning) and also the scripts/update/Updater.php at the bottom from $this->skip('7.6.0', '8.2.1'); to $this->skip('7.6.0', '8.2.2');

File: controller/Publish.php [hunk 1]
```diff
@@ -61,6 +63,19 @@ public function publish(){
             $deliveryClass = $this->getClass($classUri);
             /** @var DeliveryFactory $deliveryFactoryResources */
             $deliveryFactoryResources = $this->getServiceManager()->get(DeliveryFactory::SERVICE_ID);
+
+            $qtiTestService = $this->getServiceLocator()->get(TestModelService::SERVICE_ID);
+            $items = $qtiTestService->getItems($test);
+            $lang = \common_session_SessionManager::getSession()->getDataLanguage();
```
Comment (julien-sebire) [inline controller/Publish.php]: You can remove the non-namespaced class from the inside of the code name by using a `use` statement with an alias at the top of the file (this also gathers all dependencies together):
```
use \common_session_SessionManager as SessionManager;
...
$lang = SessionManager::getSession()->getDataLanguage();
```

File: controller/Publish.php [hunk 2]
```diff
@@ -61,6 +63,19 @@ public function publish(){
...
+
+            foreach ($items as $item) {
+                $qtiItem = QtiItemService::singleton()->getDataItemByRdfItem($item, $lang, true);
```
Comment (julien-sebire) [inline controller/Publish.php]: Please take the `QtiItemService::singleton()` part out of the `foreach` to invoke it only once.

File: controller/Publish.php [hunk 3]
```diff
@@ -25,6 +25,8 @@
 use oat\taoDeliveryRdf\model\DeliveryFactory;
 use oat\taoDeliveryRdf\model\tasks\CompileDelivery;
 use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoQtiItem\model\qti\Service as QtiItemService;
+use oat\taoQtiTest\models\TestModelService;
 use \tao_helpers_Uri;
```
Comment (julien-sebire) [inline controller/Publish.php]: FQCN in `use` statements must not have a heading `\`.
(I know you didn't modify this @k-v-vladimir )

**Comment by julien-sebire** [comment]
@k-v-vladimir Can you provide a unit test for this?

**Comment by bugalot** [comment]
I would consider to implement this while items are compiled. Indeed, doing such pre-check makes the compilation process again a little bit more heavy in terms of processing, as all items have to be fetched from DB and loaded in memory. I would prefer we raise an exception while the item is already retrieved and loaded for compilation purpose.

see taoQtiTest_models_classes_QtiTestCompiler::compileItems to explore how it could be implemented at compilation time so that we do not make extra processing

**Comment by ferenckiss89** [comment]
Not needed anymore

**Pull request closed without merging**